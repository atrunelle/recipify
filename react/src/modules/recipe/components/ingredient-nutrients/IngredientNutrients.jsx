import React, { Component } from "react";
import recipeService from "../../recipe.service";
import { Chip, Typography, Grid } from "@material-ui/core";

class IngredientNutrients extends Component {
  get allLabels () {
    const dietsLabels = recipeService.getDietLabels(this.props.nutrients.dietLabels);
    const healthLabels = recipeService.getHealthLabels(this.props.nutrients.healthLabels);
    
    return dietsLabels.concat(healthLabels);
  }
  
  get macroNutrientsData () {
    return recipeService.getMacroNutrientsList(this.props.nutrients.totalNutrients);
  }

  render() {
    if (!this.props.nutrients) return false;
    
    const nutrientsView = this.macroNutrientsData.map((nutrient, index) => {
      const quantity = Math.round(nutrient.quantity);

      return (
        <div key={index}>
          <Typography
            gutterBottom
            component="h3">
            { nutrient.label }
          </Typography>
          <Typography
            gutterBottom
            paragraph
            component="p">
              { quantity } { nutrient.unit }
            </Typography>
        </div>
      );
    });
    const allLabelsView = this.allLabels.map((label, index) => 
      <Chip key={index} label={label}/>
    );

    return (
      <Grid 
        container
        spacing={8}
        alignItems="center"
        justify="center">
        <Grid
          container 
          justify="space-between">
          {nutrientsView}
        </Grid>
        <Grid
          container
          alignItems="center"
          justify="center">
          {allLabelsView}
        </Grid>
      </Grid>
    );
  }
}
export default IngredientNutrients;