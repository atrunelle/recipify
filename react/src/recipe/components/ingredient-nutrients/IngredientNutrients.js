import React, { Component } from "react";
import recipeService from "../../../core/recipe.service";

class IngredientNutrients extends Component {

  get allLabels () {
    const dietsLabels = recipeService.getDietLabels(this.props.nutrients.dietLabels);
    const healthLabels = recipeService.getHealthLabels(this.props.nutrients.healthLabels);
    
    return dietsLabels.concat(healthLabels);
  }
  
  get macroNutrientsData () {
    return recipeService.getMacroNutrientsList(this.props.nutrients.totalNutrients);
  }

  render() {
    const nutrientsView = this.macroNutrientsData.map((nutrient) => 
      <div>
          <h3>{ nutrient.label }</h3>
          <p>{ nutrient.quantity } { nutrient.unit }</p>
      </div>
    )
    const allLabelsView = this.allLabels.map((label) => <li>{label}</li>);

    return (
      <div>
        <div className="u-text-center">
          {nutrientsView}
        </div>
        <ul>
          {allLabelsView}
        </ul>
      </div>
    );
  }
}
export default IngredientNutrients;