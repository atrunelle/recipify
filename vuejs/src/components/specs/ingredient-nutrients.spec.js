import { mount } from '@vue/test-utils';
import IngredientNutrients from '../ingredient-nutrients';

describe('Component: ingredient nutrients', () => {
  it('should match snapshot', () => {
    const wrapper = mount(IngredientNutrients, {
      propsData: {
        nutrients: {
          dietLabels: [ 'LOW_FAT' ],
          healthLabels: [ 'VEGAN' ],
          totalNutrients: {
            PROTEIN: {
              label: 'Protein',
              quantity: 2.45454,
              unit: 'gr',
            },
          },
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
