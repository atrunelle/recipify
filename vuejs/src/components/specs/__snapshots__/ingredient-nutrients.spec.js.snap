// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ingredient nutrients should match snapshot 1`] = `
<div>
  <div class="layout justify-space-between align-center row"></div>
  <div class="text-xs-center pa-2"><span tabindex="0" class="v-chip theme--light"><span class="v-chip__content">
      Low fat
    </span></span><span tabindex="0" class="v-chip theme--light"><span class="v-chip__content">
      Vegan
    </span></span></div>
</div>
`;
